### ISSUE TITLE
me/player failed: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - invalid literal; last read: 'T'

### ISSUE THREAD

**Issue by DorBreger** [kraxarn/spotify-qt#115]
### System information

Arch Linux 
XFCE
pulseaduio

### Detailed information

When running spotify-qt in terminal, it repeatedly writes
 `[10:23:52 AM] [err ] me/player failed: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - invalid literal; last read: 'T'`
When clicking the Device section in the top right, nothing happens, it doesn't even say "No devices found".
 I tried logging out and logging back in, with no success.
### What happens?

I do not know how to reproduce this issue, and I am not sure what json file it attempts to parse.
[spotify-qt-20210820.log](https://github.com/kraxarn/spotify-qt/files/7019875/spotify-qt-20210820.log)

**Comment by kraxarn** [comment]
Sounds like the API is trying to return a generic error message in plain text that fails to get parsed as valid JSON.
You could try starting spotify-qt with `--dev`, go to “Menu > Developer > Test API requests”, just enter “me”, and hit send to see if you get any more detailed error message. It will probably just say something similar, as it doesn't dump the raw response either.
Probably a good idea to dump the raw response in case of parse errors…

**Comment by DorBreger** [comment]
> Sounds like the API is trying to return a generic error message in plain text that fails to get parsed as valid JSON.
> You could try starting spotify-qt with `--dev`, go to “Menu > Developer > Test API requests”, just enter “me”, and hit send to see if you get any more detailed error message. It will probably just say something similar, as it doesn't dump the raw response either.
> Probably a good idea to dump the raw response in case of parse errors…

It works as expected again, but I will definitely try that if the error persists.

**Comment by kraxarn** [comment]
Probably just some API outage in that case. Some more detailed logging was added in 31a2f12c8eb99c0f855e4068f215cd7b45807304, but I'll keep this issue open until errors like these are better shown.

**Comment by DorBreger** [comment]
The error has returned, and I tried doing what you said, but it simply says "illegal value".
[spotify-qt-20210821.log](https://github.com/kraxarn/spotify-qt/files/7025572/spotify-qt-20210821.log)
![Screenshot_2021-08-21_16-30-18](https://user-images.githubusercontent.com/75537576/130323354-c1ab72ce-aef6-4715-944b-3c5d0df62fe1.png)
![Screenshot_2021-08-21_16-32-09](https://user-images.githubusercontent.com/75537576/130323372-74c47ed2-a67d-4a52-b832-deff89ba186c.png)

**Comment by kraxarn** [comment]
You compiled the latest version from git right? Also, enter “me” in the topmost box where the API URL is shown.

**Comment by DorBreger** [comment]
I typed it and the error it returned was` git clone --branch stable https://github.com/kraxarn/spotify-qt.git`
and yes, I did.

**Comment by kraxarn** [comment]
Ah, you need to be on the master branch, not stable. Switch with `git checkout master`, then clear your build directory and try again. You can also use the pre-built binary from [actions](https://github.com/kraxarn/spotify-qt/suites/3559341475/artifacts/85399204) if you want to.

**Comment by DorBreger** [comment]
the error still persists.

**Comment by kraxarn** [comment]
Hm, and nothing useful in the logs either? If not, I guess it will have to wait until errors are better reported, which is something planned to do very soon anyway.

**Comment by a-singer** [comment]
Just to add to this, Spotify-QT was working fine here a moment ago but is now showing exactly the same error

me/player failed: [json.exception.parse_error.101] parse error at line 1, column 1: syntax error while parsing value - invalid literal; last read: 'T'

I'm not quite sure what's going on, though I should say that when I try to go to select a device, nothing shows up. It doesn't say "no devices found" or anything like that, all I hear (I'm using a screen reader) is complete silence in that area. Normally, I would bget the name of my desktop.
I have logged out and logged back in without any change.

**Comment by a-singer** [comment]
Hi,

Just to update this, I tried a command line client with the same client authentication as Spotify-QT. I am getting "api limit exceeded, please wait x seconds) X is quite a large number, it started at 33276. I'm a bit confused about this since it always seems to happen around midnight and I only use this client, and indeed only use the account, for very limited personal playing of songs. I'm not quite sure why Spotify thinks the API limit is being exceeded, and why it has thought this around midnight eastern twice now.  I hope that's of some use.

**Comment by a-singer** [comment]
Hi,

Sorry to update this issue again, I'm not as familiar as I should be with GitHub etiquette and  apologise if I should be making a new issue or editing a previous comment. Looking into the rate limiting on Spotify, I found
https://github.com/Rigellute/spotify-tui/issues/779

I note that I have left Spotify-QT open throughout the day and that this may be causing the rate limiting through Spotify assuming that the program does send requests when music is paused as SpotifyTUI did or used to. I will make sure to close the application when I'm not actually using it and see if this error continues to happen. Otherwise, I'm not sure how Spotify-QT could possibly send so many requests for one account as to trigger this. Spotify-QT was the only application I used on this client ID before I got this error.

**Comment by kraxarn** [comment]
The app pings the API every 3 seconds by default to update playback, even when paused, but that should not be enough to reach the API limit. I've used 1 second refresh before and never had any issues with reaching the API limit, so that's very odd. Maybe a bug on Spotify's end?
Adding full logging of the API requests made is in the backlog, so that might help narrow it down.

**Comment by a-singer** [comment]
Just to update this, having started to exit  Spotify-QT when I'm not using it, I'm not getting the error in the title of this issue at all, though it's been almost a week and I got the error twice in the first three days I used the program. It's quite possible that this is, in fact, a Spotify bug, the program shouldn't be close to using too many API requests, though I do some searching as well as playing. I should say this is the only program I'm using to make API requests.
